lumibot/strategies/strategy_executor.py

```diff
@@ -161,8 +161,23 @@ def sync_broker(self):
                                 f"to be {obroker} because what we have in memory does not match the broker."
                             )
                 else:
-                    # Add to order in lumibot.
-                    self.broker._process_new_order(order)
+                    # If it is the brokers first iteration then fully process the order because it is likely
+                    # that the order was filled/canceled/etc before the strategy started.
+                    if self.broker._first_iteration:
+                        if order.status == Order.OrderStatus.FILLED:
+                            self.broker._process_new_order(order)
+                            self.broker._process_filled_order(order, order.avg_fill_price, order.quantity)
+                        elif order.status == Order.OrderStatus.CANCELED:
+                            self.broker._process_new_order(order)
+                            self.broker._process_canceled_order(order)
+                        elif order.status == Order.OrderStatus.PARTIALLY_FILLED:
+                            self.broker._process_new_order(order)
+                            self.broker._process_partially_filled_order(order, order.avg_fill_price, order.quantity)
+                        elif order.status == Order.OrderStatus.NEW:
+                            self.broker._process_new_order(order)
+                    else:
+                        # Add to order in lumibot.
+                        self.broker._process_new_order(order)
```

Comment:
With polling methods (i.e. Tradier paper trading), you can get an order here with a different state than "new" ... should it go through the same status checks as above?

Reply:
the original code had only `self.broker._process_new_order(order)`, I modified it to have the first check. i think the original code was written by you, was there a reason why you only had `self.broker._process_new_order(order)`?